src/sources/generator.rs

```diff
@@ -212,143 +214,161 @@ mod tests {
 
         let lines = &["one", "two", "three", "four"];
 
+        let mut stream = ReceiverStream::new(rx);
+
         for _ in 0..5 {
-            let event = rx.try_recv().unwrap();
+            let event = stream.next().await.unwrap();
```

Comment:
These aren't equivalent because `try_recv` will never block. I'm not entirely sure if it matters here or not, but wanted to note it just in case.

Reply:
Thanks for pointing out. I assumed this was a potential source of bugs and changed it to poll to completion.

On a second look, this indeed looks like asserting on the next poll is intentional. Will double check.

Reply:
For the record: these tests pass.

Reply:
Moved back to synchronously asserting the poll result in 149fb7926c1d888a0676d196881480ccec049c11.